### PR TITLE
test hiding the name for 50% of users on one off

### PR DESCRIPTION
**Pull request by johnduffell**
## Why are you doing this?

We don't use the first and last name for one offs, but we insist on them being entered.  This PR tests what hpapens if we remove them.

[**Trello Card**](https://trello.com/c/vU7sj6vJ/227-remove-name-fields-test-single-contributions)
BEFORE::
image
AFTER:
image

@jranks123

### CODE REVIEW

File: assets/pages/new-contributions-landing/checkoutFormIsSubmittableActions.js [hunk 1]
```diff
@@ -66,10 +67,10 @@ const getFormIsValid = (formIsValidParameters: FormIsValidParameters) => {
     firstName,
     lastName,
     email,
+    showOneOffNameFields,
   } = formIsValidParameters;
 
-  return checkFirstName(firstName)
-    && checkLastName(lastName)
+  return (!showOneOffNameFields || (checkFirstName(firstName) && checkLastName(lastName)))
```
Comment (joelochlann) [inline assets/pages/new-contributions-landing/checkoutFormIsSubmittableActions.js]: I think I'd find a ternary operator clearer here

**Comment by johnduffell** [comment]
@joelochlann I have fixed the ternary operator and added a bug fix, but I can't see name mentioned in the above file anywhere.  any hints?

**Comment by joelochlann** [comment]
you're right! I must be going mad... sorry!

**Comment by johnduffell** [comment]
not sure why that failed, I've restarted the tests

**Comment by johnduffell** [comment]
OK apprently the tests were really broken, they seem to really explode once one test falls over.  Anyway Jonny noticed the issue was real and I have fixed in https://github.com/guardian/support-frontend/pull/1305

**Pull request merged**